src/librustdoc/passes/calculate_doc_coverage.rs

```diff
@@ -31,18 +31,27 @@ fn calculate_doc_coverage(krate: clean::Crate, ctx: &DocContext<'_>) -> clean::C
 struct ItemCount {
     total: u64,
     with_docs: u64,
+    total_examples: u64,
     with_examples: u64,
 }
 
 impl ItemCount {
-    fn count_item(&mut self, has_docs: bool, has_doc_example: bool) {
+    fn count_item(
+        &mut self,
+        has_docs: bool,
+        has_doc_example: bool,
+        should_have_doc_examples: bool,
+    ) {
         self.total += 1;
 
         if has_docs {
             self.with_docs += 1;
         }
-        if has_doc_example {
-            self.with_examples += 1;
+        if should_have_doc_examples {
+            self.total_examples += 1;
+            if has_doc_example {
+                self.with_examples += 1;
+            }
```

Comment:
This will ignore variants that have examples, which doesn't seem quite right. I'm not sure what the correct behavior is though, since we don't want to _require_ examples for coverage normally.

Reply:
I can count it in case there is a doc example?

Reply:
That sounds good, although we should document that edge case since it's not intuitive.

Reply:
Agreed!